Feature: Add pre-commit step to ROS workflow

mariana-dias-alves opened the pull request:
Added a pre-commit step to the workflow for dependency installation and code checks.
Blocked by: https://github.com/MOV-AI/.github/pull/493

Tests:
-    movai_scan: https://github.com/MOV-AI/movai_scan/actions/runs/20104440461
-    movai_common: https://github.com/MOV-AI/movai_common/actions/runs/20108378814
-    movai_slam: https://github.com/MOV-AI/movai_slam/actions/runs/20108378096

mariana-dias-alves committed:
Add pre-commit tuo pipeline

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -25,6 +25,14 @@ on:
         required: false
         type: string
         default: '["release"]'
+      pre_commit:
```
not needed to have an input if you are checking the file presence dynamically

mariana-dias-alves replied:
removed

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +141,77 @@ jobs:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```
```suggestion
        uses: actions/checkout@v5
```

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +141,77 @@ jobs:
+        password: ${{secrets.registry_password}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```
```suggestion
        uses: actions/checkout@v5
```

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +141,77 @@ jobs:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install yamllint -y
```
better to install yamllint in the container `ros-buildtools`

mariana-dias-alves replied:
https://github.com/MOV-AI/containers-ros-buildtools/pull/105

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +141,77 @@ jobs:
+      - name: Run pre-commit
+        run: |
+          git config --global --add safe.directory $(pwd)
+          python -m pre_commit run --show-diff-on-failure --color=always --all-files
```
if the goal is just to check the python,  we should run only the hook that concerns python , otherwise you will run all hooks (shellcheck, bigfiles, EOF, ...)

mariana-dias-alves replied:
the goal is to run the entire static analysis

AlexFernandes-MOVAI reviewed (commented):
I have added some comments

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +141,77 @@ jobs:
+            echo "exists=false" >> $GITHUB_OUTPUT
+          fi
+
+  Pre-commit:
```
Let's use a generic name about what this step does: `Static Analysis`

mariana-dias-alves replied:
done

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +141,77 @@ jobs:
+      - name: Run pre-commit
+        run: |
+          git config --global --add safe.directory $(pwd)
+          python -m pre_commit run --show-diff-on-failure --color=always --all-files
```
what about exposing the results to the users directly on the PR

AlexFernandes-MOVAI replied:
@mariana-dias-alves 
If nothing already exist you can use something like:

```
  - name: Comment PR Results
     uses: actions/github-script@v8
      with:
        script: |
            const body = `## ....`
            github.rest.issues.createComment({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              body: body
            });
            
```

mariana-dias-alves replied:
noted and attempted

diasdm commented on .github/workflows/ros-workflow.yml:
```diff
@@ -225,6 +290,18 @@ jobs:
+      run: |
+        HOOK_ID="pylint-postbuild"
+        if grep -q "id: $HOOK_ID" .pre-commit-config.yaml; then
+          python3 -m pip install pre-commit pylint
```
Better to install these from dev-requirements

mariana-dias-alves replied:
done

mariana-dias-alves committed:
Refactor ROS workflow to remove pre_commit input

Removed pre_commit input and adjusted conditions for pre-commit checks.

mariana-dias-alves committed:
Apply suggestion from @AlexFernandes-MOVAI

Co-authored-by: Alex Fernandes <84147256+AlexFernandes-MOVAI@users.noreply.github.com>

mariana-dias-alves committed:
Rename pre-commit checks to static analysis

mariana-dias-alves committed:
Apply suggestion from @AlexFernandes-MOVAI

Co-authored-by: Alex Fernandes <84147256+AlexFernandes-MOVAI@users.noreply.github.com>

mariana-dias-alves committed:
Update ROS_BUILDTOOLS_TAG version and clean up workflow

Updated default version for ROS_BUILDTOOLS_TAG and removed unnecessary installation of yamllint.

mariana-dias-alves committed:
update pre-commit to action

mariana-dias-alves committed:
Update ros-workflow.yml

mariana-dias-alves committed:
Add checkout step to ROS workflow

mariana-dias-alves committed:
Update pre-commit action reference in workflow

mariana-dias-alves committed:
Add matrix strategy for static analysis workflow

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +137,69 @@ jobs:
+        with:
+          submodules: recursive
+      - name: Run pre-commit
+        uses: MOV-AI/.github/.github/actions/pre-commit@feature/csai-4199-1
```
just a  not to use released version of .github

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +137,69 @@ jobs:
+          args: "--color=always --show-diff-on-failure --all-files"
+          PYTHON_VERSION: ${{ inputs.PYTHON_VERSION }}
+
+      # - name: Run pre-commit
```
remove commented code

mariana-dias-alves replied:
done

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -133,20 +137,69 @@ jobs:
         distro: ${{ fromJSON(inputs.ros_distro) }}
         build_mode: ${{ fromJSON(inputs.build_modes) }}
-    if: ${{ inputs.release == 'false' }}
+    if: ${{ always() && inputs.release == 'false' && (needs.Static-analysis.result == 'success' || needs.Static-analysis.result == 'skipped') }}
```
why `always()`?

mariana-dias-alves replied:
because if pre-commit gets skipped, this step won't run without the always

mariana-dias-alves committed:
update post-build-hooks

mariana-dias-alves committed:
remove sourcing of ros repo

mariana-dias-alves committed:
Add post-build hooks to Build job

Added post-build hook execution for the Build job.

mariana-dias-alves committed:
Revert "Add post-build hooks to Build job"

This reverts commit 1c03b848856dcddbbdaf3c0770385eb03cc10492.

mariana-dias-alves committed:
move post-build-hooks and catkin test to spawner

mariana-dias-alves committed:
add root permissions to flow initiator

mariana-dias-alves committed:
remove wrong condition

mariana-dias-alves committed:
add apt update

mariana-dias-alves committed:
add missing ppa-testing

mariana-dias-alves committed:
fix ppa url

mariana-dias-alves committed:
input ROS workspce to source

mariana-dias-alves committed:
fix ROS workspace being used

mariana-dias-alves committed:
use system python for pre-commit

mariana-dias-alves committed:
move catkin tests back

mariana-dias-alves committed:
remove python config for pre-commit

mariana-dias-alves committed:
Remove PYTHON_VERSION from ros-workflow.yml

Removed PYTHON_VERSION parameter from workflow.

mariana-dias-alves committed:
Reorder Archive binary step in workflow

mariana-dias-alves commented on .github/workflows/ros-workflow.yml:
```diff
@@ -256,10 +305,52 @@ jobs:
+    
+    - name: Run post-build static analysis
+      if: ${{ needs.Check-static-analysis-config.outputs.pre_commit_exists == 'true' }}
+      uses: MOV-AI/.github/.github/actions/pre-commit@feature/csai-4199-1
```
update to release version

mariana-dias-alves committed:
update condition for post build static analysis

mariana-dias-alves committed:
Merge branch 'feature/csai-4199-v23' of github.com:MOV-AI/.github into feature/csai-4199-v23

mariana-dias-alves committed:
Change Docker image for ROS build tools

mariana-dias-alves committed:
add missing conditions to skip jobs

AlexFernandes-MOVAI committed:
Merge branch 'v2.3' into feature/csai-4199-v23

AlexFernandes-MOVAI committed:
Merge branch 'v2.3' into feature/csai-4199-v23

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -137,20 +141,66 @@ jobs:
+        with:
+          submodules: recursive
+      - name: Run static analysis
+        uses: MOV-AI/.github/.github/actions/pre-commit@feature/csai-4199-1
```
```suggestion
        uses: MOV-AI/.github/.github/actions/pre-commit@v3
```

AlexFernandes-MOVAI commented on .github/workflows/ros-workflow.yml:
```diff
@@ -260,10 +310,52 @@ jobs:
+        sudo mobros install $PACKAGES_INSTALL -y --upgrade-installed --fail-on-apt-update
+    
+    - name: Run post-build static analysis
+      uses: MOV-AI/.github/.github/actions/pre-commit@feature/csai-4199-1
```
```suggestion
      uses: MOV-AI/.github/.github/actions/pre-commit@v3
```

AlexFernandes-MOVAI committed:
Apply suggestions from code review
